presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -69,7 +68,8 @@ private JsonFunctions() {}
     public static Slice castJsonToVarchar(@SqlType(StandardTypes.JSON) Slice slice)
     {
         // TEMPORARY: added to ease migrating user away from cast between json and varchar
-        return slice;
+        throw new UnsupportedOperationException(
```

Comment:
This should throw `PrestoException` with the `NOT_SUPPORTED` error code. I would shorten the message to just say: "`CAST (jsonValue as VARCHAR)` is deprecated, use `JSON_FORMAT(jsonValue)`."